packages/hardhat-foundry/src/index.ts

```diff
@@ -0,0 +1,176 @@
+        )
+      );
+    }
+    return;
+  }
+
+  // Load foundry config
+  const foundryConfig = getForgeConfig();
+
+  // Ensure required keys exist
+  if (
+    foundryConfig?.src === undefined ||
+    foundryConfig?.cache_path === undefined
+  ) {
+    throw new HardhatFoundryError(
+      "Couldn't find `src` or `cache_path` config keys after running `forge config --json`"
+    );
+  }
+
+  // Ensure foundry src path doesnt mismatch user-configured path
+  const userSourcesPath = userConfig.paths?.sources;
+  const foundrySourcesPath = foundryConfig.src;
+
+  if (userSourcesPath !== undefined && userSourcesPath !== foundrySourcesPath) {
+    throw new HardhatFoundryError(
+      `User-configured sources path (${userSourcesPath}) doesn't match path configured in foundry (${foundrySourcesPath})`
+    );
+  }
+
+  // Set sources path
+  config.paths.sources = foundrySourcesPath;
+
+  // Change hardhat's cache path if it clashes with foundry's
+  const foundryCachePath = path.resolve(
+    config.paths.root,
+    foundryConfig.cache_path
+  );
+  if (config.paths.cache === foundryCachePath) {
+    config.paths.cache = "cache_hardhat";
+  }
+
+  pluginActivated = true;
+});
+
+// Task that transforms import names to sourcenames using remappings
+internalTask(TASK_COMPILE_TRANSFORM_IMPORT_NAME).setAction(
+  async (
+    { importName }: { importName: string },
+    _hre,
+    runSuper
+  ): Promise<string> => {
+    if (!pluginActivated) {
+      return runSuper({ importName });
+    }
+
+    const remappings = await getRemappings();
+
+    for (const [from, to] of Object.entries(remappings)) {
+      if (importName.startsWith(from) && !importName.startsWith(".")) {
+        return importName.replace(from, to);
+      }
+    }
+
+    return importName;
+  }
+);
+
+// Task that includes the remappings in solc input
+internalTask(TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE).setAction(
+  async (
+    {
+      dependencyGraph,
+      file,
+    }: {
+      dependencyGraph: DependencyGraph;
+      file: ResolvedFile;
+      solidityFilesCache?: SolidityFilesCache;
+    },
+    hre,
+    runSuper
+  ): Promise<CompilationJob | CompilationJobCreationError> => {
+    const job = (await runSuper({ dependencyGraph, file })) as
+      | CompilationJob
+      | CompilationJobCreationError;
+
+    if (!pluginActivated || isCompilationJobCreationError(job)) {
+      return job;
+    }
+
+    const remappings = await getRemappings();
+    job.getSolcConfig().settings.remappings = Object.entries(remappings).map(
+      ([from, to]) => `${from}=${to}`
+    );
+
+    return job;
+  }
+);
+
+task(
+  TASK_INIT_FOUNDRY,
+  "Initialize foundry setup in current hardhat project",
+  async (_, hre: HardhatRuntimeEnvironment) => {
+    const foundryConfigPath = path.resolve(
+      hre.config.paths.root,
+      "foundry.toml"
+    );
+
+    if (existsSync(foundryConfigPath)) {
+      console.warn(chalk.yellow(`File foundry.toml already exists. Aborting.`));
```

Comment:
This should include info to what they should do to make sure it works ok with hardhat. Possible a link to the website.

Reply:
Also, I'd throw an error instead of console and process.exit

Reply:
Something like this?

```
You've run the foundry-init task, but a foundry.toml file
already exists at /path/to/project. You don't need to
do anything else to use the hardhat-foundry plugin.

Learn more at https://hardhat.org/hardhat-and-foundry
```

And that's a shortlink to the HH and Foundry guide.